src/InfluxDB2/FluxCsvParser.php

```diff
@@ -183,10 +185,24 @@ private function addDefaultEmptyValues(FluxTable $table, $defaultValues)
     private function addColumnNamesAndTags(FluxTable $table, array $csv)
     {
         $i = 1;
-        foreach ($table->columns as &$column) {
+        $labels = array();
+        $duplicates = array();
+
+        foreach ($table->columns as $column) {
             $column->label = $csv[$i];
+            if (in_array($column->label, $labels)) {
+                $duplicates[] = $column->label;
+            } else {
+                $labels[] = $column->label;
+            }
             $i++;
         }
```

Comment:
What do you think about something more easier like:

```suggestion
        $duplicates = array();

        foreach (array_count_values($csv) as $label => $count) {
            if ($count > 1) {
                $duplicates[] = $label;
            }
            $table->columns[$i]->label = $label;
            $i++;
        }
```

?